### PR TITLE
feat(hdi,hdk): align 'mock' and 'test_utils' features

### PR DESCRIPTION
**Pull request by steveej**
### TODO:
- [x] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

**Review by thedavidmeister — APPROVED** [review]

**Review by jost-s — APPROVED** [review]

File: crates/hdk/Cargo.toml [hunk 1]
```diff
@@ -38,13 +38,14 @@ trybuild = "1.0"
...
-mock = ["hdk_derive/mock", "mockall"]
+mock = ["hdk_derive/mock", "mockall", "hdi/mock"]
 encoding = ["holo_hash/encoding"]
 fixturators = ["holochain_zome_types/fixturators", "holo_hash/fixturators"]
 test_utils = [
     "fixturators",
     "hdi/test_utils",
     "holochain_zome_types/test_utils",
     "holo_hash/test_utils",
+    "hdi/test_utils"
```
Comment (steveej) [inline crates/hdk/Cargo.toml:49]:
```suggestion
```

Comment (jost-s) [reply crates/hdk/Cargo.toml]: Why are you taking it out again?

**Comment by maackle** [comment]
I'm going through my old neglected notifications and this came up, @steveej is this something we want to merge still?

**Comment by steveej** [comment]
> is this something we want to merge still?

yes, i think it's required for mocking the HDI in app validation tests.

**Comment by thedavidmeister** [comment]
i'm gonna unsubscribe from this, it's really old

**Comment by jost-s** [comment]
@steveej Can this be closed?

**Comment by steveej** [comment]
i think at some point this will come up again, closing for now.

**Pull request closed without merging**